File: src/log.ts
import { invoke } from '@tauri-apps/api/core'

export function error(message: string) {
	invoke('log_error', { message })
}

export function warn(message: string) {
	invoke('log_warn', { message })
}

export function info(message: string) {
	invoke('log_info', { message })
}

File: src-tauri/src/log.rs
use crate::utility::Logger;

#[tauri::command]
pub fn log_info(message: &str) -> Result<(), String> {
    Logger::info(message);
    Ok(())
}

#[tauri::command]
pub fn log_error(message: &str) -> Result<(), String> {
    Logger::error(message);
    Ok(())
}

#[tauri::command]
pub fn log_warn(message: &str) -> Result<(), String> {
    Logger::warn(message);
    Ok(())
}
